### ISSUE TITLE
Fails to build on many architectures: sign-compare

### ISSUE THREAD

**Issue by nirgal** [mdbtools/mdbtools#269]
I packaged 0.9.2 on Debian experimental (Debian is in freeze and updates are currently seriously limited).

[mdbtools fails to build on many architectures](https://buildd.debian.org/status/package.php?p=mdbtools&suite=experimental), since warnings are treated as errors.

The problem is:
```
write.c: In function ‘mdb_crack_row3’:
write.c:142:14: error: comparison of integer expressions of different signedness: ‘unsigned int’ and ‘ssize_t’ {aka ‘int’} [-Werror=sign-compare]
  142 |  if (col_ptr >= mdb->fmt->pg_size || col_ptr < row_var_cols)
      |              ^~
```

**Comment by evanmiller** [comment]
Thanks for the report. The "correct" fix would be to convert pg_size to unsigned since it only takes two values (2048 and 4096). This would necessitate a change to mdbtools.h, however, and would represent an API (but not ABI?) change. A quick fix would be to cast pg_size to either `(unsigned int)` or `(size_t)` in the line cited.

**Comment by nirgal** [comment]
Agreed.
Ideally, we should have a 0.9.3 with the dirty fix, and the proper fix in the dev branch for a future 0.10.0... But I'm not willing to handle the hassle of managing this, so I lazily leave that up to you, if you want to. ^^
I think these two values, 2048 and 4096, are encoded exactly the same as signed and unsigned on all architectures, so we may even skip that, and just change ssize_t into size_t. Quick & dirty. Erk. There might be side effect everywhere the value is used, however.

**Comment by evanmiller** [comment]
Right, there might be some compile-time side effects to changing the signed-ness. Then we'd need to evaluate whether to change the return values of `mdb_read_pg` and `mdb_read_alt_pg` to unsigned also. I believe these are the only places that `ssize_t` is used in the public API. Getting rid of them may help with general portability e.g. on Windows.
